src/main/java/me/champeau/gradle/japicmp/JapicmpTask.java

```diff
@@ -168,18 +171,44 @@ private void addClasspathFor(Class<?> clazz, Set<File> classpath) {
     return toArchives(fc);
   }
 
-  private static List<JApiCmpWorkerAction.Archive> toArchives(FileCollection fc) {
+  private static List<Archive> toArchives(FileCollection fc) {
     Set<File> files = fc.getFiles();
-    List<JApiCmpWorkerAction.Archive> archives = new ArrayList<>(files.size());
+    List<Archive> archives = new ArrayList<>(files.size());
     for (File file : files) {
-      archives.add(new JApiCmpWorkerAction.Archive(file, "1.0"));
+      archives.add(new Archive(file, tryExtractVersion(file)));
     }
     return archives;
   }
 
-  private void collectArchives(final List<JApiCmpWorkerAction.Archive> archives, ResolvedDependency resolvedDependency) {
+  private static String tryExtractVersion(File file) {
+    try {
+      String name = file.getName();
+      String[] nameAndExtension = Parser.splitByLastDotChar(name);
+      String fileName = nameAndExtension[0];
+      int minusCount = fileName.contains("SNAPSHOT") ? 2 : 1;
+      int i = fileName.length();
+      while (--i >= 0 && minusCount > 0) {
+        char c = fileName.charAt(i);
+        if (c == '-') minusCount--;
+      }
+      if (minusCount == 0) {
+        return fileName.substring(i + 2);
+      }
+    } catch (Exception ignored) { }
+    return "1.0";
+  }
+
+
+  public static void main(String[] args) {
+    String s = tryExtractVersion(new File("test1-1.1.jar"));
+    assert s.equals("1.1");
+    String s1 = tryExtractVersion(new File("test1-1.2-SNAPSHOT.jar"));
+    assert s1.equals("1.2-SNAPSHOT");
+  }
```

Comment:
Please use tests if you need to check class behavior

Reply:
This is just trash which I forgot remove.